### ISSUE TITLE
Tree control focus frame is not clipped by parent container

### ISSUE THREAD

**Issue by davthedev** [godotengine/godot#74670]
### Godot version

4.1.dev (9b9bb418cb1137e69b5131ec9fa7b41c0396db28)

### System information

Ubuntu 22.04, Intel Iris Plus Graphics G7, Wayland

### Issue description

The focus frame around the Tree control overflows its parent container, even if it should be clipped due to scrolling.

![image](https://user-images.githubusercontent.com/26961646/224088478-84cf695b-b1c6-47d5-9ce1-489507bec10e.png)

### Steps to reproduce

Create a blank project and a new theme resource, to access the theme preview panel.

Focus on the tree control of the previewed widgets and scroll the whole preview to notice the overflowing frame.

### Minimal reproduction project

N/A

**Comment by Calinou** [comment]
Can you reproduce this in 3.5.2?

**Comment by davthedev** [comment]
No, it does not happens in 3.5.2, seems related with the new default theme
